bitcoin/src/taproot/mod.rs

```diff
@@ -795,6 +795,7 @@ pub struct NodeInfo {
     pub(crate) has_hidden_nodes: bool,
 }
 
+#[allow(clippy::unconditional_recursion)] // Giving a false positive.
```

Comment:
Looks like this is related: https://github.com/rust-lang/rust-clippy/issues/12181 but that one uses methods

Reply:
I spent 5 mins reading the clippy issues/PR but I'm not seeing how it applies exactly. The point you are making is that we may be able to remove this `allow` on some upcoming `clippy` release, right? That is probably true of any `allow` we use, we should periodically check them - I don't personally tend to do that ever.

Reply:
I mean if that issue gets closed we might be able to remove it (the issue is about function calls but I guess it could affect all expressions).